### ISSUE TITLE
skatOMeta freezes and produces repeated warnings

### ISSUE THREAD

**Issue by rmkoesterer** [DavisBrian/seqMeta#9]
Hi Brian,

Thanks for maintaining this project. I have an error that causes R to freeze (for at least 12 hours) when using the skatOMeta function. I pasted sessionInfo() below along with the command and warnings().

library(seqMeta)
sessionInfo()
R version 3.1.1 (2014-07-10)
Platform: x86_64-unknown-linux-gnu (64-bit)
locale:
[1] C
attached base packages:
[1] tools     stats     graphics  grDevices utils     datasets  methods  
[8] base  
other attached packages:
[1] seqMeta_1.6.0   survival_2.38-1
loaded via a namespace (and not attached):
[1] CompQuadForm_1.4.1 Matrix_1.2-1       bdsmatrix_1.3-2    coxme_2.2-5  
[5] grid_3.1.1         lattice_0.20-31    nlme_3.1-120       splines_3.1.1

ps<-readRDS("prepScoresObject.rds")
snp_info<-readRDS("snpInfo.rds")

skatOMeta(ps,SNPInfo=snp_info,rho=c(0,1),skat.wts=function(maf){dbeta(maf,1,25)},burden.wts=function(maf){as.numeric(maf<0.01)},method="saddlepoint")

Warning in pchisq(tstar \* sigmaX + muX, df = l, ncp = delta, lower.tail = FALSE) :
  full precision may not have been achieved in 'pnchisq'
Warning in pchisq(tstar \* sigmaX + muX, df = l, ncp = delta, lower.tail = FALSE) :
  full precision may not have been achieved in 'pnchisq'
Warning in pchisq(tstar \* sigmaX + muX, df = l, ncp = delta, lower.tail = FALSE) :
  full precision may not have been achieved in 'pnchisq'
Warning in pchisq(tstar \* sigmaX + muX, df = l, ncp = delta, lower.tail = FALSE) :
  full precision may not have been achieved in 'pnchisq'

... at least 50 of these

I saved the RDS files and can send them to you if you'd like a reproducible example, but I don't see that option here.

Thanks,

Ryan Koesterer
Data Analyst
Boston University School of Medicine
Department of Biomedical Genetics

**Comment by DavisBrian** [comment]
@rmkoesterer  Do you see this when you run the skatOMeta examples or only with your data set?  Also, did you see this with seqMeta 1.5?  

If you could send me a seqMeta object and snpinfo with just a gene or 2 that can reproduce the issue that would help a bunch in narrowing down the problem.

**Comment by rmkoesterer** [comment]
I sent the RDS object files to your gmail address that you posted on CRAN.

Thanks

**Comment by DavisBrian** [comment]
@rmkoesterer changing where the maf filtering is done seems to get around the problem of hanging in the example data you sent me.  I'll continue to look into why this happens but can you confirm that this works around the issue in the full data set?

```
ps<-readRDS("prepScoresObject.rds")
snp_info<-readRDS("snpInfo.rds")
skatOMeta(ps,SNPInfo=snp_info, burden.wts=1, method="saddlepoint", mafRange = c(0, 0.01), verbose=TRUE)
```

**Comment by rmkoesterer** [comment]
Thanks for looking into this so quickly!
I checked and the everything works fine when filtering for rare variants (ie. mafRange=c(0,0.01)). The one variant that gets filtered out has maf = 0.492977.

**Comment by DavisBrian** [comment]
@rmkoesterer Also, was the seqMeta object (ps) made with seqMeta 1.6?

**Comment by rmkoesterer** [comment]
Yes it was, using prepScores2. I get the same error with both seqMeta 1.5 and prepScores as well, if that helps.

**Comment by DavisBrian** [comment]
Thanks.  I'll keep looking into it.
